src/main/java/com/epam/esm/model/GiftCertificate.java

```diff
@@ -1,21 +1,74 @@
 package com.epam.esm.model;
 
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.CascadeType;
+
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
+import jakarta.persistence.Table;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
 import lombok.*;
 
-@Data
+@Getter
 @Setter
 @Builder
-@AllArgsConstructor
 @NoArgsConstructor
+@AllArgsConstructor
```

Comment:
I've deleted AllArgscontructor and Builder since I'm using a custom constructor, so I don't need them. And keep using no args constructur because it is neccesary when the database instantiates the entity, so it creates a constructor with no arguments.